lib/Basic/PlatformUtility.cpp

```diff
@@ -136,6 +136,16 @@ int sys::write(int fileHandle, void *destinationBuffer,
 #endif
 }
 
+#if defined(_WIN32)
```

Comment:
In conjunction with the typedef given above, please minimize the scope of the ifdefs to be inside the function definition